### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by I-Frostbyte**
Sum test passed.
Snapshot test failed.
Things tried so far:
- Tried installing react-test-renderer via yarn both normally and via @types/react-test-renderer
- Tried to revert back to normal js but still got same error.

Snapshot-test.tsx file is inside the utils folder.
It's test file is located within the __tests__ folder as snapshot-test-test.tsx

The environment has been upgraded for typescript jest testing following jest's website's manual configurations

This is the error given.
Screenshot from 2023-08-28 17-07-54

### CODE REVIEW

File: src/screens/main/events.screen.tsx [hunk 1]
```diff
@@ -33,13 +33,17 @@ const Events = ({navigation, route}: Props) => {
       {events ? (
         <View>
           {events?.map(event => {
+            console.log('====================================');
+            console.log({event});
+            console.log('====================================');
```
Comment (namkatcedrickjumtock) [inline src/screens/main/events.screen.tsx:38]: try to remove console.logs.

File: src/screens/main/home.screen.tsx [hunk 2]
```diff
@@ -48,9 +48,16 @@ export default function Home({navigation, route}: any) {
   //   console.log(verifiedJwt);
   // };
 
+  console.log('====================================');
+  console.log(route.params.event_id);
+  console.log('====================================');
+  console.log('====================================');
+  console.log(route.params);
+  console.log('====================================');
+
```
Comment (namkatcedrickjumtock) [inline src/screens/main/home.screen.tsx:57]: here too.

File: src/screens/main/home.screen.tsx [hunk 3]
```diff
@@ -68,6 +75,10 @@ export default function Home({navigation, route}: any) {
     getTickets();
   }, [route.params?.event_id]);
 
+  // console.log('====================================');
+  // console.log({ticketsHomeScreen: tickets});
+  // console.log('====================================');
+
```
Comment (namkatcedrickjumtock) [inline src/screens/main/home.screen.tsx:81]:
```suggestion
 
```

File: src/screens/main/scanner.screen.tsx [hunk 4]
```diff
@@ -18,7 +18,11 @@ export default function Scanner({navigation, route}: any) {
   const [purchasId, setPurchaseId] = useState();
   const [jwtCode, setJwtCode] = useState('');
 
-  const tickets = route.params?.tickets;
+  const tickets = route.params?.ticketsObj;
+
+  console.log('====================================');
+  console.log({ticketsScannerScreen: tickets});
+  console.log('====================================');
```
Comment (namkatcedrickjumtock) [inline src/screens/main/scanner.screen.tsx:25]: here

File: src/screens/main/scanner.screen.tsx [hunk 5]
```diff
@@ -58,22 +62,31 @@ export default function Scanner({navigation, route}: any) {
...
-        headers: {
-          'Content-Type': 'application/json',
-          Authorization: `Bearer ${token}`,
-        },
+    // console.log('====================================');
+    // console.log(ticket);
+    // console.log('====================================');
+    // console.log('====================================');
+    // console.log(token);
+    // console.log('====================================');
```
Comment (namkatcedrickjumtock) [inline src/screens/main/scanner.screen.tsx:70]: remove logs here too.

File: src/screens/main/scanner.screen.tsx [hunk 6]
```diff
@@ -58,22 +62,31 @@ export default function Scanner({navigation, route}: any) {
...
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token}`,
       },
-    )
+    })
       .then(data => (data.ok ? data.json() : data.json()))
       .then(data => {
+        console.log('====================================');
+        console.log({data});
+        console.log('====================================');
```
Comment (namkatcedrickjumtock) [inline src/screens/main/scanner.screen.tsx:82]:
```suggestion
 
```

File: src/screens/main/scanner.screen.tsx [hunk 7]
```diff
@@ -58,22 +62,31 @@ export default function Scanner({navigation, route}: any) {
...
         if (data.purcchase_ticket_id) {
           const ticketFound = tickets?.find(
             (t: any) => t.purcchase_ticket_id === data.purcchase_ticket_id,
           );
+          console.log('====================================');
+          console.log({ticketFound});
+          console.log('====================================');
```
Comment (namkatcedrickjumtock) [inline src/screens/main/scanner.screen.tsx:89]:
```suggestion

```

Comment (namkatcedrickjumtock) [inline src/screens/main/scanner.screen.tsx:89]:
```suggestion

```

File: src/screens/main/scanner.screen.tsx [hunk 8]
```diff
@@ -139,6 +152,11 @@ export default function Scanner({navigation, route}: any) {
       <QRCodeScanner
         onRead={({data}) => {
           verifyTicket(jwt_decode(data).ticket.purcchase_ticket_id);
+          // const decodedData: any = jwt_decode(data);
+          // console.log('====================================');
+          // console.log(decodedData);
+          // console.log('====================================');
+          // verifyTicket(decodedData.ticket.purcchase_ticket_id);
```
Comment (namkatcedrickjumtock) [inline src/screens/main/scanner.screen.tsx:159]:
```suggestion
    
```

File: src/screens/main/sync.screen.tsx [hunk 9]
```diff
@@ -47,6 +47,10 @@ const Sync = ({navigation, route}: Props) => {
   const userId = decodedToken.user_id;
   const phoneNumber = decodedToken.phone_number;
 
+  // console.log('====================================');
+  // console.log({token});
+  // console.log('====================================');
+
```
Comment (namkatcedrickjumtock) [inline src/screens/main/sync.screen.tsx:53]:
```suggestion
 
```

File: src/screens/main/sync.screen.tsx [hunk 10]
```diff
@@ -113,7 +117,7 @@ const Sync = ({navigation, route}: Props) => {
         // console.log('====================================');
```
Comment (namkatcedrickjumtock) [inline src/screens/main/sync.screen.tsx:117]:
```suggestion
       
```

**Review by namkatcedrickjumtock — COMMENTED** [review]
PR looks good, but it's a good practice not to log data inside code moving to production.
can you try to add an `Eslint` rule to flag all `console.log()` method in you're code.

**Review by namkatcedrickjumtock — APPROVED** [review]

**Review by namkatcedrickjumtock — APPROVED** [review]

**Review by Wandji69 — APPROVED** [review]
LGTM

**Pull request merged**